File: java-back-server/src/main/java/space/borisgk/taxi/api/model/dto/TripDto.java
package space.borisgk.taxi.api.model.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import lombok.*;
import space.borisgk.taxi.api.model.TripStatus;
import space.borisgk.taxi.api.model.entity.User;

import javax.persistence.*;
import java.util.Date;
import java.util.List;

@Builder
@Setter
@Getter
@AllArgsConstructor
@NoArgsConstructor
public class TripDto {
    private String id;

//    @JsonFormat(pattern = "YYYY-MM-dd'T'hh:mm:ss'Z'")
    private String date;
    private Double latFrom, latTo, longFrom, longTo;
    private TripStatus status= TripStatus.ACTIVE;
    private String addressTo, addressFrom;
    private List<UserDto> users;
    private Boolean hasFriends;
    private Integer numberOfSeats;
}

File: kotlin-back-server/src/main/kotlin/space/borisgk/taxi/api/model/dto/TripDto.kt
package space.borisgk.taxi.api.model.dto

import lombok.*
import space.borisgk.taxi.api.model.entity.User

import javax.persistence.ManyToMany
import java.util.Date

class TripDto {
    var id: Int? = null
        set(id) {
            field = this.id
        }

    var date: Date? = null
        set(date) {
            field = this.date
        }
    var fromLat: Double? = null
        set(fromLat) {
            field = this.fromLat
        }
    var toLat: Double? = null
        set(toLat) {
            field = this.toLat
        }
    var fromLong: Double? = null
        set(fromLong) {
            field = this.fromLong
        }
    var toLong: Double? = null
        set(toLong) {
            field = this.toLong
        }
    var status = TripStatus.ACTIVE
        set(status) {
            field = this.status
        }
    var toString: String? = null
        set(toString) {
            field = this.toString
        }

    var users: List<User>? = null
        set(users) {
            field = this.users
        }
}
